crates/iota-core/src/jsonrpc_index.rs

```diff
@@ -1391,40 +1367,31 @@ impl IndexStore {
     /// serves `get_AllBalance()` calls but is also used for serving
     /// `get_Balance()` queries. Notice that db read is performed with
     /// `spawn_blocking` as that is expected to block
-    pub async fn get_all_balance(
+    pub fn get_all_balance(
         &self,
         owner: IotaAddress,
     ) -> IotaResult<Arc<HashMap<TypeTag, TotalBalance>>> {
         let force_disable_cache = read_size_from_env(ENV_VAR_DISABLE_INDEX_CACHE).unwrap_or(0) > 0;
         let metrics_cloned = self.metrics.clone();
         let coin_index_cloned = self.tables.coin_index.clone();
         if force_disable_cache {
-            return spawn_blocking(move || {
-                Self::get_all_balances_from_db(metrics_cloned, coin_index_cloned, owner)
-            })
-            .await
-            .unwrap()
-            .map_err(|e| {
-                IotaError::Execution(format!("Failed to read all balance from DB: {e:?}"))
-            });
+            Self::get_all_balances_from_db(metrics_cloned, coin_index_cloned, owner).map_err(
```

Comment:
The logic is incorrect/misleading:
- if `force_disable_cache` is `false` then the cache will be used and the filter to retain non-zero balance coins be applied;
- if it's `true` then there's no such filter.